cpp/benchmarks/lists/set_operations.cpp

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <benchmarks/common/generate_input.hpp>
+#include <benchmarks/fixture/rmm_pool_raii.hpp>
+
+#include <cudf/lists/set_operations.hpp>
+
+#include <nvbench/nvbench.cuh>
+
+namespace {
+
+constexpr auto max_list_size = 20;
+
+auto generate_random_lists(cudf::size_type num_rows, cudf::size_type depth, double null_freq)
+{
+  auto builder =
+    data_profile_builder()
+      .cardinality(0)
+      .distribution(cudf::type_id::LIST, distribution_id::UNIFORM, 0, max_list_size)
+      .list_depth(depth)
+      .null_probability(null_freq > 0 ? std::optional<double>{null_freq} : std::nullopt);
+
+  auto data_table =
+    create_random_table({cudf::type_id::LIST}, row_count{num_rows}, data_profile{builder});
+  return std::move(data_table->release().front());
+}
+
+enum class SetOperations { HAVE_OVERLAP, INTERSECT_DISTINCT };
+
+void nvbench_set_op(nvbench::state& state, SetOperations op)
+{
+  cudf::rmm_pool_raii pool_raii;
+
+  auto const num_rows  = static_cast<cudf::size_type>(state.get_int64("num_rows"));
+  auto const depth     = static_cast<cudf::size_type>(state.get_int64("depth"));
+  auto const null_freq = state.get_float64("null_frequency");
+
+  auto const lhs = generate_random_lists(num_rows, depth, null_freq);
+  auto const rhs = generate_random_lists(num_rows, depth, null_freq);
+
+  state.set_cuda_stream(nvbench::make_cuda_stream_view(cudf::get_default_stream().value()));
+  state.exec(nvbench::exec_tag::sync, [&](nvbench::launch& launch) {
+    switch (op) {
+      case SetOperations::HAVE_OVERLAP:
+        cudf::lists::have_overlap(cudf::lists_column_view{*lhs}, cudf::lists_column_view{*rhs});
+        break;
+      case SetOperations::INTERSECT_DISTINCT:
+        cudf::lists::intersect_distinct(cudf::lists_column_view{*lhs},
+                                        cudf::lists_column_view{*rhs});
+        break;
+
+      default: CUDF_FAIL("Invalid set op.");
+    }
```

Comment:
Minor nit, but instead of using an enum couldn't you pass a lambda so that you don't need to define the enum or be worried about invalid ops?

Reply:
This remains TODO